### PR TITLE
remove collect and replace with map and flatten

### PR DESCRIPTION
**Pull request by ssanj**
Replacing `collect` with `map` and `flatten`. Since both `map` and `flatten` are covered before this it should be much easier and introducing something like `collect`. The only down side is explaining `swap` but that's very easy.

### CODE REVIEW

**Comment by ssanj** [comment]
If possible could peeps try and implement these solutions with `map` and `flatten`? I actually found it much easier and the exceptional various is quite verbose so it gets the point across.

File: src/main/scala/fundamentals/level03/Exceptions2EitherExercises.scala [hunk 1]
```diff
@@ -106,7 +106,8 @@ object Exceptions2EitherExercises {
     * scala> createValidPeople
     * = List(Person(Tokyo, 30), Person(Berlin, 43))
     *
-    * Hint: Use `map`, `createPerson` and `collect`
+    * Hint: Use `map` and `flatten`
+    * Hint: Use `toSeq` to convert Either into something that can be flattened with List.
```
Comment (wjlow) [inline src/main/scala/fundamentals/level03/Exceptions2EitherExercises.scala]: Can we use `toList`? Leave out `Seq` from this.

File: src/main/scala/fundamentals/level03/ExceptionExercises.scala [hunk 2]
```diff
@@ -94,16 +94,19 @@ object ExceptionExercises {
...
     * scala> createValidPeople
     * = List(Person("Tokyo", 30), Person("Berlin", 43))
     *
+    * Hint: Use `map` and `flatten`
+    *
     * What issues do you run into (if any)?
     */
   def createValidPeople: List[Person] = {
-    personStringPairs.collect {
+    personStringPairs.map {
```
Comment (wjlow) [inline src/main/scala/fundamentals/level03/ExceptionExercises.scala]: I'm worried that this derails it a little and you'll have to explain flattening a List. 

Which is okay I guess. But I'm happy to leave this up to you.

File: src/main/scala/fundamentals/level03/Exceptions2EitherExercises.scala (hunk 1 above)
Comment (ssanj) [reply src/main/scala/fundamentals/level03/Exceptions2EitherExercises.scala]: scala weirdness, disallows this. 😞

**Pull request merged**